# 인공지능수학: 통계학
# 숙제 14 (HW 14)


이름: (홍성규)

학번: (K2021035)


1. 어떤 확률 변수에 대한 확률분포가 아래 변수 $P$와 같이 주어져 있다. 이 확률분포의 엔트로피 (entropy)를 비트로 계산하시오.

In [9]:
P=[0.25, 0.1, 0.25, 0.1, 0.18, 0.12]

In [11]:
import numpy as np


In [21]:
# 여기에 코드를 작성하시오
i = [-np.log2(p) for p in P]
mul = [i[idx] * P[idx] for idx in range(len(i))]
entropy = sum(mul)
print(entropy)

2.174668276565414


(2.174668276565414 bit)

2. 아래는 2개의 확률 분포를 각각 변수 $P$와 $Q$에 저장하였다. 확률분포 $P$에 대한 확률분포 $Q$의 교차 엔트로피를 계산하시오.

In [13]:
P=[0.35, 0.25, 0.1, 0.18, 0.12]
Q=[0.32, 0.23, 0.11, 0.20, 0.14]

In [14]:
# 여기에 코드를 작성하시오
def cross_entropy(P, Q):
    return sum([-P[i] * np.log2(Q[i]) for i in range(len(P))])

print(cross_entropy(P, Q))

2.182192891170104


(2.182192891170104)